File: basic.py
'''
pip install pymongo
'''

from pymongo import MongoClient

# 방법1 - URI
mongodb_URI = "mongodb://admin:admin@127.0.0.1:27017"
client = MongoClient(mongodb_URI)

# 방법2 - HOST, PORT
# client = MongoClient(host='localhost', port=27017)

# 데이터베이스 목록
print(client.list_database_names())

# 데이터베이스 선택
db = client['testdb']

# 콜랙션 목록
print(db.list_collection_names())

# collection 선택
post_collection = db['post']

# 데이터 추가
post_collection.insert_one({"title": "title4", "author": "mung4"})
post_collection.insert_many([
  {"title": "title5", "author": "mung5"},
  {"title": "title55", "author": "mung55"},
  {"title": "title555", "author": "mung555"},
])

# 데이터 조회
print(post_collection.find_one({}))

for row in post_collection.find({}):
  print(row)

# document 수 조회

print(post_collection.count_documents({}))

File: javascript/basic.js
const Client = require('mongodb').MongoClient;

Client.connect('mongodb://admin:admin@127.0.0.1:27017/testdb', function (error, db) {
  if (error) {
    console.log(error);
  } else {
    const doc = { title: 'title1', author: 'mung1' };
    db.collection('post').insert(doc);
    db.collection('post').insertMany([
      doc,
      doc,
      doc,
    ]);
    db.close();
  }
});

Client.connect('mongodb://admin:admin@127.0.0.1:27017/testdb', function (error, db) {
  if (error) {
    console.log(error);
  } else {
    const cursor = db.collection('post').find();
    cursor.each(function (err, doc) {
      if (err) {
        console.log(err);
      } else {
        console.log(doc);
      }
    });
    db.close();
  }
});

Client.connect('mongodb://admin:admin@127.0.0.1:27017/testdb', function (error, db) {
  if (error) {
    console.log(error);
  } else {
    const cursor = db.collection('post').update(
      {title: 'title1'},
      {title: 'title1111'}
    ); // title이 title1인 document를 전부 찾아서 title을 title1111로 바꿔줌
    cursor.each(function (err, doc) {
      if (err) {
        console.log(err);
      } else {
        console.log(doc);
      }
    });
    db.close();
  }
});

Client.connect('mongodb://admin:admin@127.0.0.1:27017/testdb', function (error, db) {
  if (error) {
    console.log(error);
  } else {
    const cursor = db.collection('post').remove(
      {title: 'title1'}
    ); // title이 title1인 document를 전부 찾아서 title을 title1111로 바꿔줌
    cursor.each(function (err, doc) {
      if (err) {
        console.log(err);
      } else {
        console.log(doc);
      }
    });
    db.close();
  }
});